### ISSUE TITLE
Return keyword removed when it should not after latest update

### ISSUE THREAD

**Issue by Panajev** [nicklockwood/SwiftFormat#346]
Hello @nicklockwood ,

This seems to have changed in one of the last releases. I have not seen a new rule that would produce this kind of effects, so I would tend to think that it is a logic change inside (which is now also affecting single line closures for say... filter or map functions).

From this code:
```
        func cleanupDictionaryBlock(_ dict: (key: AnyObject, value: AnyObject)) -> Bool {
            if let valueCasted = (dict.value as? String) {
                return !valueCasted.isEmpty
            } else {
                return true
            }
        }
```

We not get:
```
        func cleanupDictionaryBlock(_ dict: (key: AnyObject, value: AnyObject)) -> Bool {
            if let valueCasted = (dict.value as? String) {
                !valueCasted.isEmpty
            } else {
                return true
            }
        }
```

Not sure if in this case the transformation is clearer or could actually be a source for bugs to come as the **method will not return when it was expected to** do (sure, in this case the answer "refactor that code now!" is also valid ;), but it does reproduce what could be considered an issue).

swiftformat, version 0.37.3

Kind Regards,

Goffredo

**Comment by nicklockwood** [comment]
@Panajev thanks for reporting. This looks like a regression in the `redundantReturn` rule, which makes sense because I did make changes in the rule.

I'll get a fix landed as soon as possible. In the meantime, you can work around it by adding `// swiftformat:disable:next redundantReturn` on the line above `return !valueCasted.isEmpty`.

**Comment by nicklockwood** [comment]
@Panajev it looks like the bug is triggered by the parentheses around `(dict.value as? String)`, so if you remove those you can avoid the issue without needing to add the swiftformat:disable directive.

**Comment by Panajev** [comment]
Thanks @nicklockwood, will use either of those workarounds before the fix on swiftformat ships :). Not sure in how many other parts of the code this would happen, I stumbled on this one tonight so it made me curious.

**Comment by nicklockwood** [comment]
@Panajev fixed in version 0.37.4

**Comment by Panajev** [comment]
Thank you @nicklockwood !
